src/main/java/com/truelayer/java/versioninfo/LibraryInfoLoader.java

```diff
@@ -29,10 +23,9 @@ public VersionInfo load() {
     private Properties getVersionInfoProperties() {
         try {
             Properties versionInfoProps = new Properties();
-            versionInfoProps.load(Files.newInputStream(Paths.get(requireNonNull(getClass()
-                            .getClassLoader()
-                            .getResource(CONFIG_FILE_PREXIF + "." + "version" + ".properties"))
-                    .getPath())));
+            versionInfoProps.load(getClass()
```

Comment:
This method directly returns an InputStream, which is suitable for reading resources from the classpath, including resources inside JAR files. This approach abstracts away the underlying file system details and works uniformly across different environments.